backend/src/routes/api/nim-serving/index.ts

```diff
@@ -1,40 +1,56 @@
 import { KubeFastifyInstance, OauthFastifyRequest } from '../../../types';
 import { createCustomError } from '../../../utils/requestUtils';
 import { logRequestDetails } from '../../../utils/fileUtils';
-
-const secretNames = ['nvidia-nim-access', 'nvidia-nim-image-pull'];
-const configMapName = 'nvidia-nim-images-data';
+import { getNIMAccount } from '../integrations/nim/nimUtils';
+import { secureAdminRoute } from '../../../utils/route-security';
 
 export default async (fastify: KubeFastifyInstance): Promise<void> => {
+  const resourceMap: Record<string, { type: 'Secret' | 'ConfigMap'; path: string[] }> = {
+    apiKeySecret: { type: 'Secret', path: ['spec', 'apiKeySecret', 'name'] },
+    nimPullSecret: { type: 'Secret', path: ['status', 'nimPullSecret', 'name'] },
+    nimConfig: { type: 'ConfigMap', path: ['status', 'nimConfig', 'name'] },
+  };
+
   fastify.get(
     '/:nimResource',
-    async (
-      request: OauthFastifyRequest<{
-        Params: { nimResource: string };
-      }>,
-    ) => {
-      logRequestDetails(fastify, request);
-      const { nimResource } = request.params;
-      const { coreV1Api, namespace } = fastify.kube;
+    secureAdminRoute(fastify)(
```

Comment:
So I believe this endpoint gets called bt the `NIMServiceModal` submit? If so, a non admin would be doing it, and reading the secrets for deployment? 

should we change `secureAdminRoute` to `secureRoute` so non admin don't have an issue?

Reply:
If you're getting regular items, you don't need to use `secureRoute` it's a misnomer... the function really should `secureJupyterRoute`

Reply:
So yeah we can revert this line back to just `async () => ...` instead of adding `secureAdminRoute`